returnn/hdf.py

```diff
@@ -75,7 +75,10 @@ def run(self):
         data = self.data
         if isinstance(data, dict):
             instanciate_delayed(data)
-            data = str(data)
+            with open("dataset.config", "wt") as dataset_file:
+                dataset_file.write("#!rnn.py\n")
+                dataset_file.write("train = %s\n" % str(data))
+            data = "dataset.config"
         elif isinstance(data, tk.Path):
             data = data.get_path()
```

Comment:
what makes you think this would not work? 

This code lives in the run function i.e. is executed in the worker. Therefore we can call `data.get_path()`. After that `data` is a str containing the location of the config file. I think providing the path to a config file is supported cf. https://github.com/rwth-i6/returnn/blob/9b8928676bdde9af93809346d6abf829d428c8b9/tools/hdf_dump.py#L102-L104
-> should work.